validator/db/kv/attester_protection.go

```diff
@@ -128,7 +128,7 @@ func (s *Store) AttestationHistoryForPubKey(ctx context.Context, pubKey [fieldpa
 				}
 				signingRoot := signingRootsBucket.Get(bytesutil.EpochToBytesBigEndian(targetEpoch))
 				if signingRoot != nil {
-					copy(record.SigningRoot[:], signingRoot)
+					record.SigningRoot = signingRoot
```

Comment:
When fetching from bolt DB, it is safest to copy, especially if this is a slice

Reply:
OK, I actually asked myself the question.
Could you explain why is should be copied?

Or maybe in this case copying is not required, but we consider it as a a safe general pattern?

Reply:
Never mind, it's in [the documentation](https://github.com/boltdb/bolt#caveats--limitations). Thanks!

> Byte slices returned from Bolt are only valid during a transaction. Once the transaction has been committed or rolled back then the memory they point to can be reused by a new page or can be unmapped from virtual memory and you'll see an unexpected fault address panic when accessing it.